### DISCUSSION TITLE
Use Gradle's convention plugin to re-structure?

### DISCUSSION THREAD

**Discussion by christopher-cudennec** [SDA-SE/sda-dropwizard-commons#1358] [Ideas]
Our main `build.gradle` is very complex and big. We could extract smaller portions into [convention plugins](https://docs.gradle.org/current/samples/sample_convention_plugins.html) that could be used like regular plugins with the submodules.

Ideas to extract to separate convention plugins:
* publishing
* everything connected to java-library
* everything connected to java-platform
* ...

In the long run I'd like to get more insights why we get warnings about dependencies between modules like the following:
```
Gradle detected a problem with the following location: '/home/runner/work/sda-dropwizard-commons/sda-dropwizard-commons/sda-commons-client-jersey-wiremock-testing/build/classes/java/main'. Reason: Task ':sda-commons-starter-example:compileTestJava' uses this output of task ':sda-commons-client-jersey-wiremock-testing:processResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

**Comment by JoergSiebahn** [comment]
I think, the key question is:
Will splitting the build configuration in dedicated files make the build easier to understand? I'm not sure about that.

And how will that tackle the dependencies? Is this really related?

**Reply by christopher-cudennec** [reply]
I'd say it will bring some better understanding to extract some common logic into a separate Gradle file.

We probably should not mix the questions about the dependencies here. I shouldn't have mentioned that here. I just found that the current Gradle setup is pretty complex when I tried to understand the warnings in the log. That's why I tried to come up with an idea how we could lower the complexity of the build files.

**Reply by JoergSiebahn** [reply]
We can give it a try but on my personal priority list tasks that help us doing the maintenance more effectively (e.g. improving the build to find green update PRs instead of manually triggering Fossa) are rated higher.

**Reply by christopher-cudennec** [reply]
A guy from Gradle support [also recommends](https://discuss.gradle.org/t/implicit-dependeny-to-subproject-in-multi-module-project/45461/6) to use convention plugin(s) instead of `subprojects` [since this is not recommended anymore](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html).
